Google.Api.Generator.Tests/ProtoTests/RoutingHeadersExplicit/Testing.RoutingHeadersExplicit/RoutingHeadersExplicitClient.g.cs

```diff
@@ -0,0 +1,146 @@
+            Modify_PlainExtractApiCall(ref _callPlainExtract);
+            _callNested = clientHelper.BuildApiCall<NestedRequest, Response>(grpcClient.NestedAsync, grpcClient.Nested, effectiveSettings.NestedSettings).WithExtractedGoogleRequestParam("table_name", new str::Regex("^(?<table_name>projects/[^/]+/instances/[^/]+/tables/[^/]+)/?$"), request => request.Nest1?.Nest2?.Name).WithGoogleRequestParam("app_profile_id", request => request.AppProfileId);
+            Modify_ApiCall(ref _callNested);
+            Modify_NestedApiCall(ref _callNested);
+            _callComplex = clientHelper.BuildApiCall<SimpleRequest, Response>(grpcClient.ComplexAsync, grpcClient.Complex, effectiveSettings.ComplexSettings).WithExtractedGoogleRequestParam("project_id", new str::Regex("^(?<project_id>projects/[^/]+)/instances/[^/]+/tables/[^/]+/?$"), request => request.Name).WithExtractedGoogleRequestParam("instance_id", new str::Regex("^projects/[^/]+/(?<instance_id>instances/[^/]+)/tables/[^/]+/?$"), request => request.Name).WithExtractedGoogleRequestParam("table_id", new str::Regex("^projects/[^/]+/instances/[^/]+/(?<table_id>tables/[^/]+)/?$"), request => request.Name).WithExtractedGoogleRequestParam("table_name", new gaxgrpc::HeaderParameterExtractor<SimpleRequest>(new str::Regex("^(?<table_name>projects/[^/]+/instances/[^/]+/tables/[^/]+)/?$"), request => request.Name).Add(new str::Regex("^(?<table_name>projects/[^/]+/instances/[^/]+/tables/[^/]+/aliases(?:/.*)?)/?$"), request => request.Name)).WithGoogleRequestParam("app_profile_id", request => request.AppProfileId).WithGoogleRequestParam("app_profile_id_renamed", request => request.AppProfileId);
+            Modify_ApiCall(ref _callComplex);
+            Modify_ComplexApiCall(ref _callComplex);
+            OnConstruction(grpcClient, effectiveSettings, clientHelper);
+            // TEST_END
+            
+            // for the easier reading, while the CR is in draft
+            _callPlainNoTemplate = clientHelper.BuildApiCall<SimpleRequest, Response>(grpcClient.PlainNoTemplateAsync, grpcClient.PlainNoTemplate, effectiveSettings.PlainNoTemplateSettings)
+                .WithGoogleRequestParam("name", request => request.Name);
+
+            _callPlainFullField = clientHelper.BuildApiCall<SimpleRequest, Response>(grpcClient.PlainFullFieldAsync, grpcClient.PlainFullField, effectiveSettings.PlainFullFieldSettings)
+                .WithExtractedGoogleRequestParam("table_name",
+                    new str::Regex("^(?<table_name>projects/[^/]+/instances/[^/]+/tables/[^/]+)/?$"),
+                    request => request.Name);
+
+            _callPlainExtract = clientHelper.BuildApiCall<SimpleRequest, Response>(grpcClient.PlainExtractAsync, grpcClient.PlainExtract, effectiveSettings.PlainExtractSettings)
+                .WithExtractedGoogleRequestParam("table_name",
+                    new str::Regex("^projects/[^/]+/instances/[^/]+/(?<table_name>tables/[^/]+)/?$"),
```

Comment:
This is where the group is important, I suspect. If the value of `request.Name` is `projects/xyz/instances/abc/tables/def` then we want to send `table_name=tables/def` in the header, yes? I *think* it would currently send the whole thing.

Reply:
👍